### DISCUSSION TITLE
ActionResponse::download with a path instead of an url?

### DISCUSSION THREAD

**Discussion by m-lotze** [laravel/nova-issues#5678] [Q&A] [answered]
Is there any reason, why `ActionResponse::download` only accepts an url and therefore only publicly available files? 

I think it should also work with a path to a file.

**Comment by crynobone** [comment, marked as answer]
Because Action are executed using XHR as default and the download link get triggered by the front-end to start the download. 

You should be able to use Storage temporary URL.
